src/orquestra/sdk/_ray/_ray_logs.py

```diff
@@ -55,6 +73,74 @@ def _update_log_filenames(self):
                     )
                 )
 
+    def _load_structured_log_record(self, record: str):
+        try:
+            data = ast.literal_eval(record)
+        except SyntaxError:
+            # Fallback option for cases where something in the record causes it not to
+            # parse correctly.
+            m = re.match(
+                "{"
+                '"timestamp": (?P<timestamp>.+), '
+                '"level": (?P<level>.+), '
+                '"filename": (?P<filename>.+), '
+                '"message": (?P<message>.+)'
+                "}",
+                record,
+            )
+
+            if m is None:
+                return
+
+            data = {
+                "timestamp": m.group("timestamp"),
+                "level": m.group("level"),
+                "filename": m.group("filename"),
+                "message": {
+                    "logs": m.group("message"),
+                    "run_id": "",
+                },
+            }
```

Comment:
**issue**: regex-based JSON parsing is a very fragile implementation. Any change in the JSON output, like a new field, or a simple swap of fields would break it. Did you consider using `json.loads()` with a `try:` + `catch (ValueError, TypeError):`?